codebuild/bin/install_s2n_head.sh

```diff
@@ -46,24 +41,27 @@ else
 fi
 set -u
 
-# Cleanup any stale s2n_head clones.
-if [[ -d "$SRC_ROOT/s2n_head" ]]; then
+s2nc_head="$DEST_DIR/s2nc_head"
+if [[ -f "$s2nc_head" ]]; then
     now=$(date +%s)
-    last_modified=$(stat -c %Y s2n_head)
+    last_modified=$(stat -c %Y "$s2nc_head")
     days_old=$(( (now - last_modified) / 86400))
-    if ((days_old > 1 )); then
-        echo "s2n_head is $days_old days old, removing and cloning again."
-        rm -rf s2n_head
-        clone
-    else
-        echo "s2n_head already exists and is $days_old days old."
+    if ((days_old <= 1)); then
+        echo "Reusing s2n_head: s2nc_head exists and is $days_old days old."
+        exit 0
     fi
-else
-    clone
 fi
-cmake "$SRC_ROOT"/s2n_head -B"$BUILD_DIR" "$EXTRA_BUILD_FLAGS" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=on
-cmake --build "$BUILD_DIR" -- -j "$(nproc)"
-cp -f "$BUILD_DIR"/bin/s2nc "$DEST_DIR"/s2nc_head
-cp -f "$BUILD_DIR"/bin/s2nd "$DEST_DIR"/s2nd_head
+
+git clone --branch main --single-branch "$CLONE_SRC" "$BUILD_DIR"
+
+cmake "$BUILD_DIR" -B"$BUILD_DIR"/build "$EXTRA_BUILD_FLAGS" \
```

Comment:
Do you actually need the BUILD_DIR argument anymore? It seems like it isn't actually necessary 🤔 or it seems like you're using it as a source dir.

Reply:
We are using it as a temporary source directory. We keep the same format we use for the real repo, so source in $BUILD_DIR and the output of cmake in $BUILD_DIR/build.

We definitely still need $BUILD_DIR. It's passed to the script, and everywhere that calls the script sets it to a random temporary folder, which is exactly where we want to put both our cloned source and our build artifacts.